src/lib/programmaticTransaction.test.js

```diff
@@ -341,4 +341,46 @@ describe('Programmatic Transaction', () => {
     querySpy.mockRestore();
     auditLogger.error.mockRestore();
   });
+
+  describe('hasModifiedData', () => {
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('returns true when changes are detected', async () => {
+      const snapshot = await transactionModule.captureSnapshot(true);
+      await db.ZALActivityRecipient.create({
+        data_id: 1,
+        dml_type: 'UPDATE',
+        old_row_data: null,
+        new_row_data: null,
+        dml_timestamp: new Date(),
+        dml_by: -1,
+        dml_as: -1,
+        dml_txid: 'cb26d433-173d-4cea-8ab0-a7af8ed37c81',
+      });
+      const result = await transactionModule.hasModifiedData(snapshot, 'cb26d433-173d-4cea-8ab0-a7af8ed37c81');
+      expect(result).toBe(true);
+    });
+
+    it('returns false when no changes are detected', async () => {
+      const snapshot = await transactionModule.captureSnapshot(true);
+      const result = await transactionModule.hasModifiedData(snapshot, 'cb26d433-173d-4cea-8ab0-a7af8ed37c81');
+      expect(result).toBe(false);
+    });
+
+    it('throws error when transaction ID is missing', async () => {
+      await expect(transactionModule.hasModifiedData([], null)).rejects.toThrow('Transaction ID not found');
+    });
+
+    it('throws error if snapshot entry is not found for a ZAL table', async () => {
+      await expect(transactionModule.hasModifiedData([], 'cb26d433-173d-4cea-8ab0-a7af8ed37c81')).rejects.toThrow('Snapshot entry not found for table: ZALActivityRecipients');
```

Comment:
I'm probably not seeing it in the git diff but is there something that sets the underlying table for the transactionModule?

Reply:
what do you mean? it is monitoring all tables in the whole database